### PR TITLE
[SPARK-30253][INFRA] Do not add commits when releasing preview version

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR add support do not add commits to master branch when releasing preview version.

### Why are the changes needed?

We need manual revert this change, example:
image

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?

manual test

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
Seems fine if it still works.

**Comment by dongjoon-hyun** [comment]
Sorry guys. I'm -1 for this.

**Comment by dongjoon-hyun** [comment]
IMO, it's just two commits which are virtually harmless.

File: dev/create-release/release-tag.sh [hunk 1]
```diff
@@ -104,7 +109,9 @@ git commit -a -m "Preparing development version $NEXT_VERSION"
 if ! is_dry_run; then
   # Push changes
   git push origin $RELEASE_TAG
```
Comment (dongjoon-hyun) [inline dev/create-release/release-tag.sh]: Where is this tag pointing, @wangyum ?

Comment (dongjoon-hyun) [reply dev/create-release/release-tag.sh]: I'm wondering if we can have a pointer during vote like the following.
- https://github.com/apache/spark/releases/tag/v3.0.0-preview-rc2

**Comment by dongjoon-hyun** [comment]
cc @gatorsmile

File: dev/create-release/release-tag.sh (hunk 1 above)
Comment (wangyum) [reply dev/create-release/release-tag.sh]: Yes. We still have a similar pointer. But we do not add any commits.

Comment (dongjoon-hyun) [reply dev/create-release/release-tag.sh]: 😮 It's interesting. Does the pointer look like the following really?
- https://github.com/apache/spark/commit/007c873ae34f58651481ccba30e8e2ba38a692c4

Comment (wangyum) [reply dev/create-release/release-tag.sh]: This is an example: https://github.com/wangyum/spark/releases/tag/v3.0.0-preview2-rc1

**Comment by wangyum** [comment]
I'm merging it and will be releasing preview2 soon.

**Comment by wangyum** [comment]
Merged to master

File: dev/create-release/release-tag.sh (hunk 1 above)
Comment (wangyum) [reply dev/create-release/release-tag.sh]: v3.0.0-preview2-rc1: https://github.com/apache/spark/commit/1b41cb5c89e1e61480bda74c16a065413813d8ed

**Pull request closed without merging**